Core/Src/state_machine.c

```diff
@@ -171,44 +150,37 @@ static int transition_nero_state(nero_state_t new_state)
 	if (!current_nero_state.home_mode && !new_state.home_mode) {
 		// Only Check if we are in pit mode to toggle direction
 		if (current_nero_state.nero_index == PIT) {
-			if (get_drive_state() == REVERSE) {
-				if (transition_drive_state(SPEED_LIMITED))
+			if (get_func_state() == REVERSE) {
+				if (transition_functional_state(F_PIT, pdu, mc))
 					return 1;
-			} else {
-				if (transition_drive_state(REVERSE))
+			} else if (get_func_state() == F_PIT) {
+				if (transition_functional_state(REVERSE, pdu,
+								mc))
 					return 1;
 			}
 		}
 	}
 
-	uint8_t max_drive_states =
-		MAX_DRIVE_STATES -
-		1; // Account for reverse and pit being the same screen
-
-	serial_print("NEW HOME MODE, %d \n", new_state.home_mode);
 	// Selecting a mode on NERO
 	if (current_nero_state.home_mode && !new_state.home_mode) {
-		if (new_state.nero_index > 0 &&
-		    new_state.nero_index < max_drive_states) {
-			if (transition_functional_state(ACTIVE))
-				return 1;
-			if (transition_drive_state(
-				    map_nero_index_to_drive_state(
-					    new_state.nero_index)))
-				return 1;
-		} else if (new_state.nero_index == OFF) {
-			if (transition_functional_state(READY))
+		if (new_state.nero_index < DEBUG &&
+		    new_state.nero_index > OFF) {
+			if (transition_functional_state(new_state.nero_index,
+							pdu, mc))
 				return 1;
 		}
 	}
 
 	// Entering home mode
 	if (!current_nero_state.home_mode && new_state.home_mode) {
-		if (transition_functional_state(READY))
+		if (transition_functional_state(READY, pdu, mc))
 			return 1;
 	}
 
 	cerberus_state.nero = new_state;
+	/* Notify NERO */
+	osThreadFlagsSet(nero_monitor_handle, NERO_UPDATE_FLAG);
```

Comment:
we want to continously send out messages, mqtt is a lossy network, and so is CAN